README.md
# prizm.dev
An info website for me

[view](https://voxelprismatic.github.io/prizm.dev)


text/changes.txt
#] feb 27, 2020
___
> Starting up fixed
> > Now, if there is an issue with starting up, it still starts up fully instead of \
stopping all commands from loading. Also, it says that it's broken in [#is-prizm-online]\
<https://discordapp.com/channels/533290351184707584/556247032701124650>

> The react command now supports numbers too
> > Eg `;]react t r y 1 m 8`
===

#] feb 15, 2020
___
> Backups are now easily made, just by using a simple command.
> > Only I, PRIZ, can use this command as it contains PRIZM's token, which is the username, \
password, and other things I'm sure.
> > It uploads directly to discord, and that's why only I can use it.
> The command docs can easily be made too, and only I can use it because I made it that way.
===

-~-

#] feb 09, 2020
___
> The `;]graph` command now supports polars!
> > *Don't graph too many or it won't graph some properly... idk why
===

-~-

#] jan 27, 2020
___
> Guild join bug fixed
> > When PRIZM joins a guild, it now actually has a prefix for that guild instead of throwing \
an error constantly
===

-~-

#] dec 30, 2019
___
> The AI command has been updated, and no longer supports symbols because it doesn't under \
stand them in the first place
===

-~-

#] dec 28, 2019
___
> All math commands have been refreshed with new RegEx
> > Also all math equations are now parsed by the same parser
> > All math commands now have their answers unparsed as well so `1j` will be `1i`

> Added new statuses

> Fixed a bug where none of the math commands actually worked

> Still working on the new `PRIZMATIC` module
===

-~-

#] nov 30, 2019
___
> Logging was fixed yet again
> > Edited messages now works, but roles supposedly don't

> Added the `;]triangle` command
> > Solves your triangles

> The graph command has been updated
> > More colors and themes

> Some math commands have been updated
> > Better RegEx for equations and things
===

-~-

#] oct 30, 2019
___
> Logging now #ACTUALLY# works!!!!
> > The issue was that the `bot` logging was actually a switch to turn the entire system off
    
> Updated most logging things
> > Now is cleaner and always attaches a file [when available] with additional context
===

-~-

#] oct 29, 2019
___
> The following commands have been fixed and work for all mods
> > `;]pre`
> > `;]mng`
> > `;]dsbl`
> > `;]enbl`
> > `;]audit`
===

-~-

#] oct 27, 2019
___
> Updated the `;]graph` command
> > Now parses quickly
> > Now won't break when commands contain `e`
> > Now shouldn't break when commands have `x(5x-3)` or similar
> > Now can "emulate" calculators
> > > Honestly too many ill just paste the source code instead online
> > > Emulation only changes colors, screen ratio/size, and graph step
> > Added asymptote detection, add `--det` or `--asym` to the end of the equation

> All commands have been updated
> > They now have better documentation for non techies
> > > `INT` and `FLOAT` have been changed to `NUMBER`
> > > `STR` has been changed to `TEXT`
> > > `ROLE`, `MESSAGE`, `MEMBER`, `CHANNEL`, and more provide info on how to obtain/input that data
> > > `[NO ARGS...` has been changed to `[NO INPUT...`

> Renamed a few commands
> > `;]emj` --> `;]ei`
> > `;]gld` --> `;]gi`
> > `;]msg` --> `;]ti`
> > `;]chnl` --> `;]ci`
> > `;]rol` --> `;]ri`
> > `;]usr` --> `;]ui`
> > `;]mbr` --> `;]mi`

> Fixed a few commands
> > `;]rng` and `;]coin` now actually work

> Added the `;]colors` command
> > Displays a random color or displays a sample of a hex color provided
===

-~-

#] oct 21, 2019
___
> Added the `;]convert` command!
> > You can now convert your 4mb files to any format you like!
> > What it currently supports ---
> > > Image ] `PNG`, `GIF`, `JPG`, and most other common image formats
> > > Audio ] `MP3`, `M4A`, `WAV`, `AIFF` and all other `FFmpeg` formats
> > > Video ] `MP4`, `OGV`, `M4V`, and all other `FFmpeg` formats
> > Syntax ] `;]convert {output format} <attach a file to the message>`
> Use the `;]bug` command to report any issues. 
> One is known ] FFmpeg doesn't always work
===

-~-

#] oct 18, 2019
___
> Added many more faces like `;]` and `;O`

> Added many more status texts

> Updated the `;]char` command to have bytes for python
===

-~-

#] oct 17, 2019
___
> The `;]draw` command is now a thing!
> > Draw whatever you want in a 128x128 grid!
> > Will be upscaled to a 1024x1024 image when sending to prevent blurriness
```
#] HELP FOR DRAW [CATAGORY: fun]
=] Have fun drawing with others!
>  USAGE - ';]draw {?color} {?x1} {?y1} {?x2} {?y2} ... {?xN} {?yN}'
>  ALIAS - "paint", "color", "canvas", "painting", "drawing", "coloring"
COLOR  [STR] - The color you want to place, MUST BE HEX
xN, yN [ANY] - The XY coordinates [MAX - 127, MIN - 0]
- - You can use ranges like 0-16
- - You can be relative like ~1
- - You can use relative ranges like ~9-26
*If no params are passed, the image is sent
*If only coordinates are passed, the color at that coordinate is sent along with the image
*ALL relative points are relative to the previously drawn point IN THE SAME MESSAGE
*^ Any starting relative points will be relative to 0, 0
```
===

-~-

#] oct 16, 2019
___
> Added the `;]draw` command
> > Still working on it, and will be greatly improved tomorrow
===

-~-

#] oct 14, 2019
___
> The char command now gives more info
> > When requesting one character, it tells you how to write it in many languages
> > > Includes PY, JS, URI/URL, JAVA, C#, C++, C, HTML, RUBY, and SWIFT
> > > If you want other languages, please visit the +[\#prizm\_suggest]<https://discordapp.com/channels/533290351184707584/535507316775059516/> channel
===

-~-

#] oct 13, 2019
___
> The `;]reddit` command has been updated
> > Now properly states whether or not the post is archived
> > Now states whether or not the `[IMAGE]` button should be pressed
===

-~-

#] oct 11, 2019
___
> The `;]reddit` command has been updated
> > Now looks much cleaner
> > Attributes---
> > > `[-] ` - Archived
> > > `O-O ` - Link submission
> > > `TXT ` - Text submission

ti84ce/emu.js
/**
 * NOTICE --------
 * TI Does not endorse this content, and this isn't an emulator.
 * This file only creates pixel elements and prints them to the page.
 * All "emulators" are made by hand in TI-BASIC, then converted to JS
 * accordingly.
 */

var graphScreen = [];
var fullScreen = [];

var inGraph = false;

function randInt(mn, mx) {
    return Math.floor(Math.random() * Math.abs(mx - mn)) + Math.min(mn, mx);
}

function range(mn = 0, mx = null, step = 1) {
    var ls = [];
    if(mx == null) {
        mx = mn;
        mn = 0;
    }
    for(var x = mn; x < mx; x += step)
        ls.push(x);
    return ls;
}

//Colors

let BLUE =     10;
let RED =      11; 
let BLACK =    12;
let MAGENTA =  13;
let GREEN =    14;
let ORANGE =   15;
let BROWN =    16;
let NAVY =     17;
let LTBLUE =   18;
let YELLOW =   19;
let WHITE =    20;
let LTGRAY =   21;
let MEDGRAY =  22;
let GRAY =     23;
let DARKGRAY = 24;

let OFFGRAY =   1;
let MINT =      2;
let OFFBLUE =   3;
let OFFWHITE =  4;

//Make Screen

for(var y of range(165)) {
    graphScreen.push([]);
    for(var x of range(265)) {
        graphScreen[graphScreen.length - 1].push(BLACK);
    }
}

 //emulate();

function drawScreen() {
    var borderColor = randInt(OFFGRAY, OFFWHITE);
    for(var y of range(240)) {
        fullScreen.push([]);
        for(var x of range(320)) {
            if(y <= 30)
                fullScreen[fullScreen.length - 1].push(DARKGRAY);
            else if(inGraph)
                fullScreen[fullScreen.length - 1].push(borderColor);
            else
                fullScreen[fullScreen.length - 1].push(WHITE);
        }
    }
    
    if(inGraph) {
        for(var y of range(165))
            for(var x of range(265))
                fullScreen[y + 32][x + 63] = graphScreen[y][x];
    }

    var screen = "";

    for(var y of fullScreen) {
        screen += "<div>";
        for(var x of y) {
            screen += `<div class="ti${x}"></div>`;
        }
        screen += "</div>";
    }
    
    setHtml("ti84ce", screen);
}
